components/infobox/wikis/stormgate/infobox_league_custom.lua

```diff
@@ -0,0 +1,250 @@
+local RaceBreakdown = Lua.import('Module:Infobox/Extension/RaceBreakdown', {requireDevIfEnabled = true})
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Breakdown = Widgets.Breakdown
+local Cell = Widgets.Cell
+local Center = Widgets.Center
+local Title = Widgets.Title
+
+local CustomLeague = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _league
+
+local FALLBACK_DATE = '2999-99-99'
+
+---@param frame Frame
+---@return Html
+function CustomLeague.run(frame)
+	local league = League(frame)
+	_league = league
+	_args = league.args
+
+	_args.game = Game.name{game = _args.game}
+	_args.liquipediatiertype = _args.liquipediatiertype or _args.tiertype
+	_args.raceBreakDown = RaceBreakdown.run(_args) or {}
+	_args.player_number = _args.raceBreakDown.total
+	_args.maps = CustomLeague._getMaps(_args)
+	_args.number = Logic.isNumeric(_args.number) and string.format('%05i', tonumber(_args.number)) or nil
+	--varDefault because it could have been set above the infobox via a template
+	_args.status = _args.status or Variables.varDefault('tournament_status')
+	_args.cancelled = CustomLeague._checkCancelled(_args)
+	_args.finished = CustomLeague._checkFinished(_args)
+
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
+	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
+	league.addToLpdb = CustomLeague.addToLpdb
+	league.liquipediaTierHighlighted = CustomLeague.liquipediaTierHighlighted
+
+	return league:createInfobox()
+end
+
+---@param args table
+---@return boolean
+function CustomLeague._checkCancelled(args)
+	return args.status == 'cancelled' or Logic.readBool(_args.cancelled or Variables.varDefault('cancelled tournament'))
+end
+
+---@param args table
+---@return boolean
+function CustomLeague._checkFinished(args)
+	if _args.cancelled then
+		return true
+	end
+	if args.status == 'finished' then
+		return true
+	end
+
+	local finished = Logic.readBoolOrNil(args.finished)
+	if finished ~= nil then
+		return finished
+	end
+
+	local queryDate = _league:_cleanDate(args.edate) or _league:_cleanDate(args.date) or FALLBACK_DATE
+	if os.date('%Y-%m-%d') >= queryDate then
+		local data = mw.ext.LiquipediaDB.lpdb('placement', {
+			conditions = '[[pagename::' .. string.gsub(mw.title.getCurrentTitle().text, ' ', '_') .. ']] '
+				.. 'AND [[opponentname::!TBD]] AND [[placement::1]]',
+			query = 'date',
+			order = 'date asc',
+			limit = 1
+		})
+		if data and data[1] then
+			return true
+		end
+	end
+
+	return false
+end
+
+---@return WidgetInjector
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	if id == 'gamesettings' then
+		table.insert(widgets, Cell{name = 'Game Version', content = {CustomLeague._getGameVersion(_args)}})
+	elseif id == 'customcontent' then
+		if _args.player_number and _args.player_number > 0 then
```

Comment:
```suggestion
		if Logic.isNumeric(_args.player_number) and tonumber(_args.player_number) > 0 then
```

Reply:
is already numeric due to processing via lines 47/48 + `Module:RaceBreakdown`